readers/api/transport.go

```diff
@@ -207,25 +209,34 @@ func encodeError(_ context.Context, err error, w http.ResponseWriter) {
 	}
 }
 
-func authorize(r *http.Request, chanID string) error {
-	token := r.Header.Get("Authorization")
-	if token == "" {
-		return errors.ErrAuthentication
-	}
-
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
-	defer cancel()
-
-	_, err := auth.CanAccessByKey(ctx, &mainflux.AccessByKeyReq{Token: token, ChanID: chanID})
-	if err != nil {
-		e, ok := status.FromError(err)
-		if ok && e.Code() == codes.PermissionDenied {
-			return errors.ErrAuthorization
+func authorize(ctx context.Context, req listMessagesReq, tc mainflux.ThingsServiceClient, ac mainflux.AuthServiceClient) (err error) {
+	switch {
+	case strings.HasPrefix(req.token, userToken):
+		token := strings.TrimPrefix(req.token, userToken)
+		user, err := usersAuth.Identify(ctx, &mainflux.Token{Value: token})
+		if err != nil {
+			e, ok := status.FromError(err)
+			if ok && e.Code() == codes.PermissionDenied {
+				return errCannotAuthorizeUser
+			}
+			return errCannotAuthorizeUser
+		}
+		_, err = thingsAuth.IsChannelOwner(ctx, &mainflux.ChannelOwnerReq{Owner: user.Email, ChanID: req.chanID})
+		if err != nil {
+			e, ok := status.FromError(err)
```

Comment:
Move this line of initialization below, in `if` statement.

Reply:
For code readability, I would maybe keep it this way. I leave it to @mteodor to decide.

Reply:
I would leave it as it is